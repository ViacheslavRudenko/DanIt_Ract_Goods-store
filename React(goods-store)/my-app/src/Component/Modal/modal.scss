.modal {
  z-index: 1;
  position: fixed;
  overflow: auto;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.05);

  &__box {
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);

    &--item {
      border-radius: 10px;
      background-color: white;
      max-width: 600px;
      min-width: 420px;
      color: black;
    }
  }

  &__header {
    display: flex;
    justify-content: space-between;
    background-color: aliceblue;
    padding: 15px 30px;
    font-weight: bold;
    font-size: 24px;

    .header__title {
      margin: 0 auto;
      padding-top: 15px;
    }

    .btn {
      &__item {
        border: none;
        font-size: larger;
        color: black;
        font-weight: 100;
        background-color: inherit;
      }
    }
  }

  .main {
    padding: 20px 50px;
    &__btn {
      display: flex;
      justify-content: center;
      width: 100%;
    }
  }

  .goods {
    display: flex;
    padding-bottom: 20px;
    &__img {
      max-width: 100px;
    }
    &__info {
      padding-left: 15px;
    }
  }
  .info {
    &__title {
      font-style: italic;
    }
    &__price {
      text-align: end;
      font-weight: bold;
      padding-top: 10px;
      display: flex;
      justify-content: space-between;
    }
  }
}
