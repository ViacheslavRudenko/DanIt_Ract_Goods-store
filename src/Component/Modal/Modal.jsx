import React from "react";
import "./modal.scss";
import Button from "../Button/Button";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";

export default function Modal({ closeModal, actionWithModal }) {
  const modal = useSelector((store) => store.modal);
  const { title, btn, isBtnClose, content } = modal;

  return (
    <div className="modal">
      <div className="modal__box">
        <div className="modal__box--item">
          <div className="modal__header ">
            <p className="header__title">{title}</p>
            {isBtnClose && (
              <div className="header__btn btn">
                <button className="btn__item" onClick={closeModal}>
                  X
                </button>
              </div>
            )}
          </div>
          <div className="modal__main main">
            {content}
            <div className="main__btn btn">
              <Button btn={btn} actionWithModal={actionWithModal} />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

Modal.propTypes = {
  closeModal: PropTypes.func,
  actionWithModal: PropTypes.func,
};
