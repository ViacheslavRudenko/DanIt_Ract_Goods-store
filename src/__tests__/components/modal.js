import Modal from "../../Component/Modal/Modal";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { createStore } from "redux";
import reducer from "../../store/reducers/modal/modal";
import renderer from "react-test-renderer";

const product = {
  Id: "selinazawacki-shirt",
  Maker: "selinazawacki",
  img: "https://user-images.githubusercontent.com/41929050/61567058-142c1c80-aa33-11e9-89fb-b4f30d84d69d.png",
  Url: "https://www.instagram.com/p/BEXlpiZCnJ3",
  Title: "Floppy Crop",
  Description: "Used up the Diskette fabric today to make 2 of these crops.",
  Ratings: null,
  article: "3c260",
  price: 23000,
  color: "blue",
};

const store = createStore(reducer, {
  modal: { btn: [{ id: 1, text: "Add to cart" }], title: "TITLE" },
});
const onChange = jest.fn();
const Component = (
  <Provider store={store}>
    <Modal closeModal={onChange} actionWithModal={onChange} product={product} />
  </Provider>
);

describe("<Modal/> component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render component", () => {
    render(Component);
    screen.getByRole("button");
  });

  it("should render text in component", () => {
    render(Component);
    screen.getByText(/TITLE/);
    screen.getByText(/Floppy Crop/i);
    screen.getByText(/3c260/i);
    screen.getAllByText(/23000/);
  });

  it("should pass OnClick if btn is clicked", () => {
    render(Component);
    userEvent.pointer({
      keys: "[MouseLeft]",
      target: screen.getByRole("button"),
    });
    expect(onChange).toHaveBeenCalled();
  });
});
