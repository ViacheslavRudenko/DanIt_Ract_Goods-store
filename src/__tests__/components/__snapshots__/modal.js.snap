// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> component renders correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal__box"
  >
    <div
      className="modal__box--item"
    >
      <div
        className="modal__header "
      >
        <p
          className="header__title"
        >
          TITLE
        </p>
      </div>
      <div
        className="modal__main main"
      >
        <div
          className="main__text goods"
        >
          <img
            alt=""
            className="goods__img"
            src="https://user-images.githubusercontent.com/41929050/61567058-142c1c80-aa33-11e9-89fb-b4f30d84d69d.png"
          />
          <div
            className="goods__info info"
          >
            <p
              className="info__title"
            >
              Floppy Crop
            </p>
            <p
              className="info__price"
            >
              <span>
                Article: 
                3c260
              </span>
               Price:
               
              23000
            </p>
          </div>
        </div>
        <div
          className="main__btn btn"
        >
          <button
            className="active"
            id={1}
            onClick={[MockFunction]}
          >
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
